configure.ac

```diff
@@ -76,6 +76,15 @@ AS_IF([test "x$have_glog" = "xno"],
 
 save_LIBS="$LIBS"
 AS_UNSET([LIBS])
+AC_SEARCH_LIBS([snappy_compress], [snappy],,, [$save_LIBS])
```

Comment:
This is the tricksy bit: if you have `libsnappy-dev` installed, it will detect it, and `leveldb_LIBS` will have `-lleveldb -lsnappy`. If you have the older `libleveldb-dev` from Ubuntu Precise that only has static libraries, this is what you have to do. If you have the newer one that has shared libraries (where `libleveldb.so` is linked with `libsnappy.so`, so you don't have to), then it's spurious, but shouldn't harm anything.

I figure this is an acceptable compromise, keeping things working, in exchange for having an extra direct dependency that we have anyway, indirectly.